File: Delivery System 2/warehouse.cpp
#include "warehouse.h"

// Конструктор
Warehouse::Warehouse(int id, const Address& address, const string& phoneNumber)
	: m_id(id), m_address(address), m_phoneNumber(phoneNumber) {
}

// Прием заказа на склад
void Warehouse::receiveOrder(Order* order) {
	m_storedOrders.push_back(order);
}

File: Delivery System 2 Java/src/Warehouse.java
package src;

import java.util.ArrayList;
import java.util.List;

public class Warehouse {
    private int id;
    private Address address;
    private String phoneNumber;
    private List<Order> storedOrders;

    // Конструктор
    public Warehouse(int id, Address address, String phoneNumber) {
        this.id = id;
        this.address = address;
        this.phoneNumber = phoneNumber;
        this.storedOrders = new ArrayList<>();
    }

    // Методы
    public void receiveOrder(Order order) {
        storedOrders.add(order);
    }

    // Геттеры
    public int getId() { return id; }
    public Address getAddress() { return address; }
    public String getPhoneNumber() { return phoneNumber; }
}